src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2695,7 +2695,9 @@ trait Types
   private final class ClassArgsTypeRef(pre: Type, sym: Symbol, args: List[Type]) extends ArgsTypeRef(pre, sym, args)
   private final class AliasNoArgsTypeRef(pre: Type, sym: Symbol) extends NoArgsTypeRef(pre, sym) with AliasTypeRef
   private final class AbstractNoArgsTypeRef(pre: Type, sym: Symbol) extends NoArgsTypeRef(pre, sym) with AbstractTypeRef
-  private final class ClassNoArgsTypeRef(pre: Type, sym: Symbol) extends NoArgsTypeRef(pre, sym)
+  private final class ClassNoArgsTypeRef(pre: Type, sym: Symbol) extends NoArgsTypeRef(pre, sym) {
+    override final def contains(sym: Symbol): Boolean = this.sym == sym
```

Comment:
should could also consider the prefix

Reply:
No prob! This PR is looking awesome.

Reply:
How would you write it to apply @retronym suggestion? I have reverted that part, to see if it compiles.